Docker

LakesideMiners opened the pull request:
See below

mistressAlisi committed:
Initial commit

mistressAlisi committed:
initial ply

mistressAlisi committed:
Ply Galleries now display cards properly. Updated Gallery serialisers. Initial work on Gallery display

mistressAlisi committed:
Gallery and ingestor now work with Celery

mistressAlisi committed:
First MAJOR Ply update for 2022
-Introducing DynaPages and Widgets for Rendering
-Introducing the Profile rendering engine based on DynaPages
-Introduced the first DynaPage Widget API and sample APIs.
-Upgraded Gallery Serialisers
-Updated Fields to Profiles, added RP stats: HP/MP/STA
-Added gender and species fields to profiles
-Introduced groups (nee Clans), title and member systems
-Gallery uploader with Celery+Rabbit is now working
-Frameworks are coming into place for Ply 1.0..

mistressAlisi committed:
New Dynapage Widgets
Community Page template with working login
Vhost Community implemented
Basic login and entrance to dashboard and profile complete

mistressAlisi committed:
Added a whole bunch of modules:
-stats
-combat
-skills
-equipment
-spells
-items
-forge

-Started work on (WorldBuilder) forge and Character  / profile creation
-Javascript tools and fixes

mistressAlisi committed:
New Ply 2022.021801 Version:
-Dynapage registry extended for wigdets
-Introducing widget registry json file
-Dynapage init routines
-Magic UUIDs
-Magic/System Profiles
-Initial Templates for Profiles
-Introducing Forge character creation process (working)
-Avatar uploading working through forge
-Dashboard now supports multiple character profiles cleanly
-Ported all Avatar images to new Forge-compatible API (Testdata is probably no longer required)

mistressAlisi committed:
Version fix

mistressAlisi committed:
Several bugfixes regarding Dynapage and profile creation fixed.

mistressAlisi committed:
-Much work on Dynapages,
-New JSON format for widget IDs.
-JSON ingestion tool.

-New Widgets:
	-Health Display (HPDisplay)
	-Stats Display (StatsDisplay)
-Forge now creates stats in chars

mistressAlisi committed:
Partial commit, initial work on almanac, new community admin api, new work on dashboard login/logout bit, dashboard user profile switch now supports multiple module redirect.

mistressAlisi committed:
-Added Almanac Models for Martor pages for markup.
-Renamed user to public views in Alamanc.
-Introduced community admin profiles
-introduced almanac editor tools permissions based on Admin profiles.
-First introduction of public profile viewer at /profiles/[profile_id] or /profiles/@[profile_id].

mistressAlisi committed:
Tiny commit to enable UWSGI

mistressAlisi committed:
Tiny commit to enable UWSGI

mistressAlisi committed:
-NEW Gallery MODAL Viewer that NOW WORKS!!
-Gallery Photo plugin now renders to Viewer
-Fixes to Views and create category to include community parameters
-Added new "dashboard_generic_avatar" files for generic quickie avatars anywhere.
-Ported the Dashboard templates to the quickie avatar api
-renamed gallery_core self-references to 'this'

mistressAlisi committed:
-Gallery modal viewer now fully working.
-Collections now viewable.
-Gallery modal viewer key bindings for navigation enabled.
-Gallery viewer counters and updaters for media views.
-CSS Fixes
-New API Endpoint for counters, /gallery/api/view_counter/item to update items AND collections using GET requests

mistressAlisi committed:
-Forge: Now able to Edit active profile.
-Forge: Linked edit profile urls to javascript
-Forge: Modified templates for forms to support profile editing
-Dashboard: Added Forge entry for profile editing
-Community: implemented 'restricted' flag.

mistressAlisi committed:
-GALLERY: View now exports permission's community and community uuid
-GALLERY: Serialiser now can serialise all of a community's collecctions
-GALLERY: decoupled profiles from the serialisers
-GALLERY: gallery_core.js: updated api paths to be /absolute for now
-GALLERY: public gallery is implemented and viewable
-GALLERY: views are now supported (counter)
-GALLERY: implemented public urls into plycore

mistressAlisi committed:
-GALLERY: Implemented AWS S3 Storage API
-GALLERY: Fixed view counter
-GALLERY: Implemented deletion support for items
-GALLERY: Ported file paths to pathlib.Path api
-PLY.UTILITIES: Ported upload file paths to pathlib.Path api

mistressAlisi committed:
PLY v.2022.04.2200:
-***Almanac:
-Added Links for Forge Community tools in the menu for community admins.

-****Community:
-Added support for background items from the gallery
-Added editing community support
-****Dashboard:
-Updated Navmenu
-Introducing new 'profileid interactions widget'
-Updated Sidebar to reflect new version code from global ply.settings

-***Dynapages:
-Added CSS updates for community cover background support
-Added background support as above ;)

-****Forge:
-Added support for Community editing

-***Gallery:
-Added global collection view
-Added per-profile views
-Added per-profile/per-collection views

-***PLY[core]:
-Added global version support
-Added global hostname support
-File uploader improved

-***Profiles:
-Introduced 100px and shorthand profile header class

mistressAlisi committed:
Added requirements.txt

mistressAlisi committed:
-Cleaned up Migrations from Django 3.0.2 to Django 4.0 to enable installation in new nodes.
-Almanac forms race condition when database not installed fixed.

mistressAlisi committed:
Minor modifications to Dynapage Api for installation ease

mistressAlisi committed:
Minor fix to let communities work without background assignment.

mistressAlisi committed:
Added debug flag

mistressAlisi committed:
Intermediate commit to freeze HTML templates

mistressAlisi committed:
Fixed Javascript loader to prevent race condition in Gallery_photos when using DO/S3. Removed user login on api call /api/items/get to enable gallery functionality w/o login.

mistressAlisi committed:
Enabling Metrics

mistressAlisi committed:
Enabling Metrics for Gallery items with remote addr

mistressAlisi committed:
Enabling Metrics for Gallery items with remote addr - bugfix to data model

mistressAlisi committed:
Metrics bugfixes

mistressAlisi committed:
Metrics bugfixes

mistressAlisi committed:
Metrics bugfixes

mistressAlisi committed:
****METRICS:
-Basic metrics extended to profiles, galleries and communities.
-Introduced the Metrics toolkit.

mistressAlisi committed:
Metrics bugfixes

mistressAlisi committed:
Metrics bugfixes

mistressAlisi committed:
Metrics bugfixes

mistressAlisi committed:
Metrics bugfixes

mistressAlisi committed:
Metrics: Enable Gallery homepage hits

mistressAlisi committed:
Metrics: Enable Gallery homepage hits

mistressAlisi committed:
Metrics: Enable Gallery homepage hits

mistressAlisi committed:
Metrics: Enable Gallery homepage hits

mistressAlisi committed:
Metrics: Bugfix for profile metrics.

mistressAlisi committed:
Metrics: Bugfix for profile metrics.

mistressAlisi committed:
Metrics: Bugfix for profile metrics.

mistressAlisi committed:
Simple commit to make Gallery collections required; quality of life UI improvements for the lightbox for review and submit.

mistressAlisi committed:
***ADMIN/AUTH/LOGIN:
-Registration is now supported.
-Registration uses django-registration and default engine.
-Registration now supports password change and email activation.
-Initial templates for registration forms.
-Intial templates for registration emails.
***FORGE:
-Initial profile creation for new accounts supported
-Profile selector introduced
****Dashboard/Global:
-Login/Logout links
-Register link
-Profile switcher now works

mistressAlisi committed:
updated requirements txt for latest commit

mistressAlisi committed:
***ALMANAC:
- Introducing Page Editor Form
- Introducing Menu Editor UI And Forms
- Cleaned up User Views
***COMMUNITY:
- Moved friends and followers to community
- New Functions and views are now needed for Ply to work!!
***DASHBOARD:
- Added html5sortable plugin
- Dynapage login widget fixes
- Gallery core now sets page metadata for OG (open graph)
- Templates updated to expose OG Metadata.
- Reworked navbar
***NOTIFICATIONS:
-Introducing notifications API
-Introducing notifications objects/views
***PROFILES:
-Ported friends and followers to Community API
***STREAM:
-Initial Stream API and models
****PLY:
-Now with basic openGraph metadata!

mistressAlisi committed:
***GALLERY:
-Introduced new, properly working Share URL.
-New Share URL now uses PLY sharing redirect basic button.
-Viewer now has Share buttons and interactions.
-New API urls for metrics.
-Viewer now has CLOSE button hanging over everything.
***METRICS:
-Introduced 'Share viewed' metric.
-Introduced 'Shared' metric.
-Implemented these two metrics for galleries :).
***PLY:
-Introduced sharing card/redirect link basic template.

mistressAlisi committed:
Fixing community front page queryset error

mistressAlisi committed:
Improvements and bugfixes to opengraph share metadata

mistressAlisi committed:
Improvements and bugfixes to opengraph share metadata

mistressAlisi committed:
Improvements and bugfixes to opengraph share metadata

mistressAlisi committed:
Improvements and bugfixes to opengraph share metadata

mistressAlisi committed:
Improvements and bugfixes to opengraph share metadata

mistressAlisi committed:
Improvements and bugfixes to opengraph share metadata

mistressAlisi committed:
Improvements and bugfixes to opengraph share metadata

mistressAlisi committed:
CSS fix

mistressAlisi committed:
Modifing share url

mistressAlisi committed:
Reworking Sharing views to support bots

mistressAlisi committed:
Added meta:twitter sharing tags to share view

mistressAlisi committed:
Sharing widget fixes

mistressAlisi committed:
Sharing widget fixes

mistressAlisi committed:
Sharing widget fixes

mistressAlisi committed:
Sharing widget fixes

mistressAlisi committed:
Sharing widget fixes

mistressAlisi committed:
Sharing widget fixes

Commit:
hydrus support

mistressAlisi committed:
Merge pull request #1 from LakesideMiners/main

hydrus support

LakesideMiners committed:
Update README.md

mistressAlisi committed:
Merge pull request #2 from LakesideMiners/LakesideMiners-readme-pretty

Update README.md

mistressAlisi committed:
Added new Readme.MD; thank you @smolKitters \ Transitioned settings to CSV for vhosts

mistressAlisi committed:
Adding Streams API and initial migration... needed to move forward.

mistressAlisi committed:
GALLERY:
moved gallery max upload size to settings
SETTINGS:
NEW SETTING: 'PLY_GALLERY_MAX_ORIGINAL_SIZE' : max original size in kb, adjust .env accordingly.

mistressAlisi committed:
***Starting Documentation with Sphinx!
***Added Emoji application and support
***emoji support in streams
***stream work... not done yet.
***gallery: now with garbage collector
***ply toolkit: introducing reqtools

mistressAlisi committed:
Documentation fix

mistressAlisi committed:
Requirements cleanup, removed old django-celery

mistressAlisi committed:
adding YAML file for readthedocs.io

mistressAlisi committed:
adding YAML file for readthedocs.io

mistressAlisi committed:
LOTS of work on Streams and SQL. Introducing SLhud API points, lots of work...

mistressAlisi committed:
Initial support for Gallery items in Streams, stream URL fixes, gallery thumbnail filter

mistressAlisi committed:
***** MASSIVE WIP: ***
- Dynapages:
	- Introducing Dynapages '2.0':
        - Reworked most of the API. Dynapage templates now live inside foobar/templates/dynapages,
        - Dynapages now support dynawidgets/pages from -any- app.
        - Dynapages tools were reworked for this modification. (register profiles again if you already have a db)
- Dashboard:
	- Ported APIs to DynaPages as above
	- Fixed NavMenu and broken dropdowns
	- Sidebar fixed (collapse)
	- Introduced preferences module
- Community:
	- Updated for Dynapages 2.0
- Gallery:
	- Introduced Editing and management API for collections and items
	- Updated Dashboard rendering of collections
	- Added edit/copy/delete/move forms for collection control
	- Updated serialiser to expose archived property
	- Ported counters to new SQL trigger interface
	- introduced Dashboard Management CSS and JS
	- Introduced "Card API" for creating plugin-driven editing interfaces for items
	- Enhancements to the modal viewer.
	- Introduces the 'thumbnailcard' Template filter
	- Introduces the 'Gallery Simple Frame Viewer' Widget
- Notifications:
	- Introduced Notifcations App and API
	- Introduced notification URLs to Dashboard and Application Menu
	- Introduces Notifications SQL functions
	- Introduces Mentions
-PLYCORE:
	- Cleaned up teamplate 'ply-bootstrap5-stickyfooter' and 'ply-bootstrap5'
	- TOOLKIT: File uploader now records original file name
- Preferences:
	- Introducing Preferences App
	- Introduces Preferences Forms
	- Introduces Views for Dashboard
	- Introduces save/edit/reset behaviours and API endpoints
	- Introduces preferences Datamodel tied to USER.
-STREAM:
	- Ported to Dynapages 2.0
	- Introduces Stream 'Latest Casts Widget'
	- Updates to Community and profile view
	- Cast animation in CSS now working

mistressAlisi committed:
**PART OF Massive WIP still in progress**
- Renamed SQL files to consistent naming schema nnn_name.sql.
- introduced install_sql_files management command.

mistressAlisi committed:
--- Still Massive WIP!
- Added Categories App (just skeleton rn)
- Dashboard:
  *Added SmartSelect Plugin
  *Cleaned up and updated Selectize
 - Gallery:
   * Added APIs for View/Edit settings and Details.
   * Added Metadata/File view.
   * Added delete submission and delete temp file support.
   * Updated serialiser to support active/inactive flag.
   * Introduced many more parts of the card_api for gallery: details_Form/settings_form
   * Introduced gallery collection manager
- Gallery_photos:
  * Adding Forms and workflow for Editing submissions
  * Some bugfixes.

mistressAlisi committed:
-MASSIVE WIP IS BEING WRAPPED UP:
- Introducing Categories App with 'Discipline' and 'Category'
- Introducing Categories command line tools to load Disciplines and Categories.
- Added sample files for disciplines and categories to DOCUMENTATION.
- Gallery: Items now must have a 'Category' applied to them.
- Gallery: Added several item flag columns for API and feature completion
- Gallery: Added Review/Update Detail/Setting API using Plugin infrastructure
- Gallery/tasks: Introduced two new tasks for managing Galleries, including update_gallery_items.
- Toolkit: Can now get original files, too! Not just upload them!
- SETTINGS: For DEBUG Purposes: we have introduced the 'ALWAYS_LOAD_S3' flag.

mistressAlisi committed:
SQL fixes to notifications functions

mistressAlisi committed:
SQL fixes to notifications functions

mistressAlisi committed:
SQL fixes to notifications functions

mistressAlisi committed:
Added Commit category support in publish tasks.

mistressAlisi committed:
Gallery debug code

mistressAlisi committed:
Gallery debug code

mistressAlisi committed:
Gallery debug code

mistressAlisi committed:
Gallery Task cleanup

mistressAlisi committed:
Gallery Task cleanup

mistressAlisi committed:
Gallery Task cleanup

mistressAlisi committed:
Gallery Task cleanup

mistressAlisi committed:
Fixes keyword creation for new keywords in cellery gallery tasks

Commit:
Fixes Publish

Commit:
fixes upload cleaner settings

mistressAlisi committed:
Fixes keyword creation for new keywords in cellery gallery tasks

mistressAlisi committed:
Fixes keyword creation for new keywords in cellery gallery tasks

mistressAlisi committed:
Fixes keyword creation for new keywords in cellery gallery tasks

mistressAlisi committed:
Debugging celery back and forth

mistressAlisi committed:
Fixes to the Gallery Models and background processor

mistressAlisi committed:
Fixes to the Gallery Models and background processor

mistressAlisi committed:
Fixes to the Gallery Models and background processor

mistressAlisi committed:
PLY Gallery:
-Introduced basic Image zoom in support for Photos.

mistressAlisi committed:
Gallery zoom JS fix

mistressAlisi committed:
Gallery zoom JS fix

mistressAlisi committed:
SLAGENT:
-Fixes the Bridge SLUUIDs
Settings:
-Deploy Grappelli.

mistressAlisi committed:
PLY SETUP:
- Introducing PLY Setup tools and scripts.
- settings.py updated to support setup.sh
- Introduced 'create_community' command.
- New Requirements.txt - Ply really looks nice with Grappelli :)
DECOUPLE:
- Decouple now supports the 'config/settings.ini' file besides .env.
SLHUD:
- SLHUD now has auth, registration, and login working fully.
- SLHUD uses dynapages :3

LakesideMiners committed:
init

LakesideMiners committed:
Merge branch 'mistressAlisi:main' into docker

LakesideMiners committed:
over 5 hours

LakesideMiners committed:
next up, feed a rabbit some Celery

LakesideMiners committed:
Update Dockerfile

LakesideMiners commented on ply/settings.py:
```diff
@@ -116,12 +116,12 @@
-        'PASSWORD': config('DB_PW'),
-        'HOST': config('DB_HOST'),
-        'PORT': config('DB_PORT')
+        'ENGINE': 'django.db.backends.postgresql',
```
changed

LakesideMiners commented on ply/settings.py:
```diff
@@ -116,12 +116,12 @@
-        'PASSWORD': config('DB_PW'),
-        'HOST': config('DB_HOST'),
-        'PORT': config('DB_PORT')
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': os.environ.get('POSTGRES_NAME'),
+        'USER': os.environ.get('POSTGRES_USER'),
+        'PASSWORD': os.environ.get('POSTGRES_PASS'),
+        'HOST': 'db',
+        'PORT': 5432,
```
Changed to take input from docker and use static port so it is matched up with what docker needs.

mistressAlisi replied:
Any specific reason we can't use dynamic ports using config or os.environ?

LakesideMiners reviewed (commented):
In addation to getting started on Docker support, settings.py had these changes made to allow for it.

LakesideMiners committed:
Update settings.py

part 1/2 
should work, just adds another value that needs to be set, but, it can be worked with,
